xdsl/transforms/eqsat_create_eclasses.py

```diff
@@ -0,0 +1,79 @@
+from xdsl.context import MLContext
+from xdsl.dialects import builtin, eqsat, func
+from xdsl.ir import Block
+from xdsl.passes import ModulePass
+from xdsl.pattern_rewriter import (
+    GreedyRewritePatternApplier,
+    PatternRewriter,
+    PatternRewriteWalker,
+    RewritePattern,
+    op_type_rewrite_pattern,
+)
+from xdsl.rewriter import InsertPoint, Rewriter
+from xdsl.utils.exceptions import DiagnosticException
+
+
+def insert_eclass_ops(block: Block):
+    # Insert eqsat.eclass for each operation
+    for op in block.ops:
+        results = op.results
+        if len(results) != 1:
+            raise DiagnosticException("Ops with non-single results not handled")
+
+        eclass_op = eqsat.EClassOp(results[0])
+        insertion_point = InsertPoint.after(op)
+        Rewriter.insert_op(eclass_op, insertion_point)
+        results[0].replace_by(eclass_op.results[0])
+        # Redirect eclassop operand back to the original value
+        eclass_op.operands[0] = results[0]
+
+    # Insert eqsat.eclass for each arg
+    for arg in block.args:
+        eclass_op = eqsat.EClassOp(arg)
+        insertion_point = InsertPoint.at_start(block)
+        Rewriter.insert_op(eclass_op, insertion_point)
+        arg.replace_by(eclass_op.results[0])
+        # Redirect eclassop operand back to the original value
+        eclass_op.operands[0] = arg
```

Comment:
I opened a PR for the helper you suggested, should work here #3196

Reply:
Lovely, will use that instead